packages/lib-classifier/src/store/SubjectStore.js

```diff
@@ -112,13 +112,14 @@ const SubjectStore = types
     }
 
     function append (newSubjects) {
-      newSubjects.forEach(subject => {
-        self.resources.put(subject)
-      })
+      const validResources = isValidReference(() => self.resources)
```

Comment:
`isValidReference` is for checking `types.safeReference` or `types.reference` types (see: https://github.com/mobxjs/mobx-state-tree#reference-validation-isvalidreference-tryreference-oninvalidated-hook-and-typessafereference). `resources` are `types.map` (which has a built in `types.optional` with a default), so this isn't necessary and might not work? 

Instead, I think you can:

```
if (self.resources && self.resources.size > 0) {
  // do stuff with self.resources
}
```

Reply:
Looking at this again, I think `self.resources.put(subject)` was failing the `isStateTreeNode` check here so maybe `subject` is invalid? I'm not sure. https://github.com/mobxjs/mobx-state-tree/blob/9121976541e5392a5364c3cdf3d046f7f0165b8d/packages/mobx-state-tree/src/types/complex-types/map.ts#L172

Reply:
To examine the error, run `newSubjects.forEach(self.resources.put)` outside the `if` block.

Reply:
Checking the map size is greater than 0 will stop us adding subjects to an empty queue.

Reply:
Doh, you're right. `if (self.resources)` should be enough to check for existence then.

Reply:
Hopefully, but the error is thrown inside `self.resources.put` so there may need to be an additional check that it’s a valid tree node.

Reply:
From those docs,
>In addition to the options possible for a plain reference type, the optional options parameter object also accepts a parameter named acceptsUndefined, which is set to true by default, so it is suitable for model properties. When used inside collections (arrays/maps) it is recommended to set this option to false so it can't take undefined as value, which is usually the desired in those cases.

So I think arrays and maps should be updated as per the example there eg. 
```js
resources: types.map(types.safeReference(Subject, { acceptsUndefined: false }))
```
I don't think that will fix the error encountered here but it seems like good practice.